test/run-tests

```diff
@@ -867,17 +902,63 @@ def main():
     # default values for config
     config = {"repositories": [], "images": [], "name": "linux-system-roles-test"}
 
+    # check for settings via env. vars
+    args.secrets = os.environ.get("TEST_HARNESS_SECRETS", args.secrets)
+    args.config = os.environ.get("TEST_HARNESS_CONFIG", args.config)
+    args.cache = os.environ.get("TEST_HARNESS_CACHE", args.cache)
+    args.inventory = os.environ.get("TEST_HARNESS_INVENTORY", args.inventory)
+    args.pull_request = os.environ.get("TEST_HARNESS_PULL_REQUEST", args.pull_request)
+    args.use_images = os.environ.get("TEST_HARNESS_USE_IMAGES", args.use_images)
+    args.dry_run = os.environ.get("TEST_HARNESS_DRY_RUN", args.dry_run)
+
     with open(args.config + "/config.json") as configfile:
         config.update(json.load(configfile))
         images = config["images"]
         random.shuffle(images)
         repos = [r.split("/") for r in config.get("repositories", [])]
         random.shuffle(repos)
 
+    # this is the name for github status reporting
+    config["name"] = os.environ.get("TEST_HARNESS_CONFIG_NAME", config["name"])
+    env_repos = os.environ.get("TEST_HARNESS_REPOSITORIES")
+    if env_repos:
+        repos = [r.split("/") for r in env_repos.split(",")]
     setup_logging(config.get("logging", {}))
 
     check_environment()
 
+    ansible_version = LooseVersion(get_ansible_version())
+    logging.info("Using Ansible version {}".format(ansible_version))
+    # this is used in PR status - context name
+    ansible_ver_str = (
+        f"ansible-{ansible_version.version[0]}.{ansible_version.version[1]}"
+    )
+
+    image_patterns = args.use_images.split(",")
+    test_images = []
+    for pattern in image_patterns:
+        for image in images:
+            min_ansible_version = image.get(
+                "min_ansible_version", "0"
+            )
+            if ansible_version < LooseVersion(min_ansible_version):
+                logging.info(
+                    f"Skipping image {image['name']} - ansible version "
+                    f"{ansible_version} is less than the minimum ansible "
+                    f"version {min_ansible_version} for this image"
+                )
+                continue
+            if fnmatch.fnmatch(image["name"], pattern) or fnmatch.fnmatch(
+                image["source"], pattern
+            ):
+                if image not in test_images:
+                    test_images.append(image)
+            else:
+                logging.info(
+                    f"Skipping image {image['name']} - name and/or source "
+                    f"do not match image pattern {pattern}"
```

Comment:
This seems to be more like a debug message. Also it is misleading, since if there are multiple patterns and only one matches an image, there are multiple messages that the image is skipped but it is actually used (which is not logged). To be more correct, all candidates that were not selected need to be compared at the end against all images that were selected and then a message could be logged (but this would also not log images that were skipped because of the ansible version).

Reply:
fixed in https://github.com/linux-system-roles/test-harness/pull/87/commits/a1f7e36ec876eb927bb6ddcd25ba619b13dc88c0